File: shifai-ios/ShifAI/App/AppConfig.swift
import Foundation

/// Centralized app configuration.
/// All environment-specific values in one place.
enum AppConfig {

    // MARK: - Supabase

    static let supabaseURL = ProcessInfo.processInfo.environment["SUPABASE_URL"]
        ?? "https://your-project.supabase.co"

    static let supabaseAnonKey = ProcessInfo.processInfo.environment["SUPABASE_ANON_KEY"]
        ?? "your-anon-key"

    // MARK: - Feature Flags

    static let enableMLPredictions = true
    static let enableCloudSync = true
    static let enableWidgets = true
    static let enableShareExport = true
    static let enableBiometric = true

    // MARK: - Thresholds

    static let minCyclesForML = 3
    static let maxNotificationsPerDay = 1
    static let quietHoursStart = 22  // 22:00
    static let quietHoursEnd = 7     // 07:00
    static let autoStopIgnoreThreshold = 3
    static let syncRetryLimit = 3
    static let encryptionKeyLength = 256

    // MARK: - NFR Targets

    static let maxAppLaunchMs: Double = 2000
    static let maxSyncLatencyMs: Double = 3000
    static let maxMLInferenceMs: Double = 500
    static let targetCrashFreeRate: Double = 99.5
    static let minAccessibilityScore: Double = 90

    // MARK: - URLs

    static let privacyPolicyURL = URL(string: "https://shifai.app/privacy")!
    static let termsOfServiceURL = URL(string: "https://shifai.app/terms")!
    static let supportURL = URL(string: "https://shifai.app/support")!

    // MARK: - Analytics

    static let analyticsEndpoint = "https://plausible.io/api/event"
    static let analyticsDomain = "shifai.app"

    // MARK: - Storage

    static let maxExportSizeMB = 10
    static let shareLinkExpiryHours = 24
    static let databaseName = "shifai_encrypted.db"
}

File: shifai-android/app/src/main/java/com/shifai/app/AppConfig.kt
package com.shifai.app

/**
 * Centralized app configuration.
 * All environment-specific values in one place.
 */
object AppConfig {

    // ─── Supabase ───
    var supabaseURL: String = "https://your-project.supabase.co"
        private set
    var supabaseAnonKey: String = "your-anon-key"
        private set

    fun initializeSupabase(url: String, key: String) {
        supabaseURL = url
        supabaseAnonKey = key
    }

    // ─── Feature Flags ───
    const val ENABLE_ML_PREDICTIONS = true
    const val ENABLE_CLOUD_SYNC = true
    const val ENABLE_WIDGETS = true
    const val ENABLE_SHARE_EXPORT = true
    const val ENABLE_BIOMETRIC = true

    // ─── Thresholds ───
    const val MIN_CYCLES_FOR_ML = 3
    const val MAX_NOTIFICATIONS_PER_DAY = 1
    const val QUIET_HOURS_START = 22
    const val QUIET_HOURS_END = 7
    const val AUTO_STOP_IGNORE_THRESHOLD = 3
    const val SYNC_RETRY_LIMIT = 3
    const val ENCRYPTION_KEY_LENGTH = 256

    // ─── NFR Targets ───
    const val MAX_APP_LAUNCH_MS = 2000L
    const val MAX_SYNC_LATENCY_MS = 3000L
    const val MAX_ML_INFERENCE_MS = 500L
    const val TARGET_CRASH_FREE_RATE = 99.5
    const val MIN_ACCESSIBILITY_SCORE = 90.0

    // ─── URLs ───
    const val PRIVACY_POLICY_URL = "https://shifai.app/privacy"
    const val TERMS_OF_SERVICE_URL = "https://shifai.app/terms"
    const val SUPPORT_URL = "https://shifai.app/support"

    // ─── Analytics ───
    const val ANALYTICS_ENDPOINT = "https://plausible.io/api/event"
    const val ANALYTICS_DOMAIN = "shifai.app"

    // ─── Storage ───
    const val MAX_EXPORT_SIZE_MB = 10
    const val SHARE_LINK_EXPIRY_HOURS = 24
    const val DATABASE_NAME = "shifai_encrypted.db"
}
